server/server.go

```diff
@@ -129,13 +138,21 @@ func (a *ArgoCDServer) Run() {
 	}
 
 	// Start the muxed listeners for our servers
-	log.Infof("argocd %s serving on port %d (tls: %v, namespace: %s)", argocd.GetVersion(), port, a.useTLS(), a.Namespace)
+	log.Infof("argocd %s serving on port %d (url: %s, tls: %v, namespace: %s, sso: %v)",
+		argocd.GetVersion(), port, a.settings.URL, a.useTLS(), a.Namespace, a.settings.IsSSOConfigured())
 	go func() { errors.CheckError(grpcS.Serve(grpcL)) }()
 	go func() { errors.CheckError(httpS.Serve(httpL)) }()
 	if a.useTLS() {
 		go func() { errors.CheckError(httpsS.Serve(httpsL)) }()
 		go func() { errors.CheckError(tlsm.Serve()) }()
 	}
+	if a.settings.IsSSOConfigured() {
+		go func() {
+			time.Sleep(2 * time.Second)
```

Comment:
Is sleep required to make sure that side container has started? Looks like we are waiting for dex in `registerDexHandlers`:
```
func (a *ArgoCDServer) registerDexHandlers(mux *http.ServeMux) {
   dexClient.WaitUntilReady()
}
```

Can we remove sleep?

Reply:
This is an interesting chicken and egg problem which stems from ArgoCD being an OIDC client to itself. Initially we waited for dex to come up, but that was just so we could create the client app entry in dex. 

Whats happening here, is that we need the OIDC provider to be up and running in order to initialize the client app. The OIDC client library we are using, wants to HTTP GET http://argocd.com/api/dex/.well-known/openid-configuration. This cannot happen until the http server is up and running. 

The sleep is because i noticed that after starting the http server, it wouldn't respond immediately. I actually want to change sleep to a retry so that it's not so arbitrary.

I'll also include an explanation as comment.